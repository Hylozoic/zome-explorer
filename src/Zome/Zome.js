import React, { useState } from 'react'
import ZomeFunction from '../ZomeFunction'
import ExpandButton from '../ExpandButton'

export default function Zome ({ zome, callZome }) {
  const { name, description, fn_declarations } = zome
  const callZomeFunc = callZome(zome.name)  

  const [expanded, setExpanded] = useState(false)
  const toggleExpanded = () => setExpanded(!expanded)  

  return <div className='zome'>
    <div onClick={toggleExpanded} className='clickable-div'>
      <div className='zome-name'>
        {name}
        <ExpandButton expanded={expanded} toggleExpanded={toggleExpanded} />
      </div>
      {expanded && <div className='zome-description'>{description}</div>}
    </div>
    {expanded && <div className='zome-functions'>
      <div className='zome-functions-header'>Functions</div>
      {fn_declarations.map(fnDeclaration => 
      <ZomeFunction         
        fnDeclaration={fnDeclaration} 
        callZomeFunc={callZomeFunc} 
        zomeName={name}
        key={fnDeclaration.name} />)}
    </div>}
  </div>
}