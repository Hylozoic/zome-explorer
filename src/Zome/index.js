import Zome from './Zome'
export default Zome