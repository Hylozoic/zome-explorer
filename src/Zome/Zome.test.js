import Zome from './Zome'
import { shallow } from 'enzyme'
import React from 'react'

it('matches the last snapshot', () => {
  const props = {
    zome: {
      name: 'some zome',
      description: 'a really great one',
      fn_declarations: [
        {name: 'a function declaration'},
        {name: 'a different function declaration'},        
      ]
    },
    callZome: name => () => {},    
  }  
  const wrapper = shallow(<Zome {...props} />)
  expect(wrapper).toMatchSnapshot()
})
