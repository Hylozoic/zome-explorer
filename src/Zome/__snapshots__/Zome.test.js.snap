// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the last snapshot 1`] = `
<div
  className="zome"
>
  <div
    className="clickable-div"
    onClick={[Function]}
  >
    <div
      className="zome-name"
    >
      some zome
      <ExpandButton
        expanded={false}
        toggleExpanded={[Function]}
      />
    </div>
  </div>
</div>
`;
